src/main/java/lesser/citibike/CitiBikeHelper.java

```diff
@@ -0,0 +1,115 @@
+package lesser.citibike;
+
+import io.reactivex.rxjava3.core.Single;
+import io.reactivex.rxjava3.schedulers.Schedulers;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class CitiBikeHelper {
+    private final CitiBikeService service;
+    private Single<StationsResponse> stationsResponse;
+    private Single<StatusResponse> statusResponse;
+
+    public CitiBikeHelper(CitiBikeService service) {
+        this.service = service;
+    }
+
+    private Single<StationsResponse> getStations() {
+        if (stationsResponse == null) {
+            stationsResponse = service.getStations().subscribeOn(Schedulers.io());
+        }
+        return stationsResponse;
+    }
```

Comment:
This doesn't work the way you think it does if you call it twice. 

I'm impressed that you're trying to use operators like zip and map, but this is not the way that I showed you how to use retrofit.